src/routes/users/sign-up-route.js

```diff
@@ -1,37 +1,72 @@
-const qs = require('querystring');
+const fs = require("fs");
+const path = require("path");
 
 const saveUser = user => {
-  // получить файл с юзером
-  // найти путь папки users
-  // сохранить туда файл
+  const userName = user.username;
+  const fileUser = path.join(
+    __dirname,
+    "../../",
+    "db",
+    "/users",
+    `${userName}.json`
+  );
+  fs.writeFile(fileUser, JSON.stringify(user), function(err) {
+    if (err) throw err;
+    console.log(`${userName}.json was created`);
+  });
 };
 
-const signUpRoute = (request, response) => {
-  // Взять данные что пришли
+const checkUser = user => {
+  const userName = user.username;
+  const userPhone = user.telephone.replace(/\s/g, "");
+  const userPass = user.password;
+  const userEmail = user.email;
+  if (
+    typeof userName === "string" &&
+    !isNaN(Number(userPhone)) &&
+    typeof userPass === "string" &&
+    typeof userEmail === "string"
+  )
+    return true;
+  else return false;
+};
 
-  if (request.method === 'POST') {
-    let body = '';
+const signUpRoute = (request, response) => {
+  if (request.method === "POST") {
+    let body = "";
 
-    request.on('data', function (data) {
+    request.on("data", function(data) {
```

Comment:
It's better to move body aggregation logic into separate function. Even now it's hard to read code because of a lot of tabulations.